schema/resource.go

```diff
@@ -125,13 +125,43 @@ func (resource *Resource) Update(updateData map[string]interface{}) error {
 	}
 	for _, property := range resource.schema.Properties {
 		id := property.ID
+
 		if val, ok := updateData[id]; ok {
-			data[id] = val
+			newVal, err := updateResourceRecursion(val, data[id])
+			if err == nil {
+				data[id] = newVal
+			} else {
+				return err
+			}
 		}
 	}
 	return nil
 }
 
+func updateResourceRecursion(updateData interface{}, sourceData interface{}) (res interface{}, err error) {
+	switch updateData.(type) {
+	case map[string]interface{}:
+		newUpdate, foundUpdate := updateData.(map[string]interface{})
```

Comment:
at this point you already know that `updateData` is a `map[string]interface{}`, so additional conversion is not needed. you could use `switch t := t.(type) {` syntax to simplify it. (cf. https://golang.org/doc/effective_go.html#type_switch)